src/Core/backpropagation/BackupNegamax.cpp

```diff
@@ -2,14 +2,20 @@
 
 BackupNegamax::BackupNegamax(){};
 
-void BackupNegamax::backup(std::shared_ptr<SearchNode> node, Reward defaultPolicyScore, int currentPlayer) {
+void BackupNegamax::backup(const std::shared_ptr<SearchNode> &node, Reward defaultPolicyScore, int currentPlayer) {
```

Comment:
What was the reason to make this a reference if it is already defined as a pointer